### DISCUSSION TITLE
Bar chart or Pie Chart & Interactive

### DISCUSSION THREAD

**Discussion by Saupia** [teamalgoritma/community#416] [Q&A] [answered]
(1) Saya membuat satu barplot untuk skala ketahanan bahasa. Tetapi bar tidak menampilkan informasi secara interactive padahal saya sudah running ggplotly. 
(2) [minta saran] Saya mencoba menggunakan pie chart untuk jenis file ini karena saya pikir saya hanya perlu presentasikan informasi secara umum. Punya saran lain?

````{r}
lang_stat <- 
ggplot(data = freq_egids, mapping = aes(x = freq, 
                                        y = reorder(EGIDS_Label, freq))) +
  geom_col(mapping = aes(fill = freq)) +
  scale_fill_gradient(low = "red", high = "black") +
    labs(title = 'Total of languages in Indonesia based on the endangerment scale' ,
       x = 'Total of languages',
       y = 'Endangerment level') +
  theme_minimal() +
  theme(legend.position = "none") 
```
```{r}
ggplotly(p = lang_stat, tooltip = "text")
````

![image](https://user-images.githubusercontent.com/114889462/198191238-1393e42e-43fa-4e48-8730-586beb0311a5.png)

**Comment by ranydwi** [comment, marked as answer]
Dear Pak Renhard @Saupia ,

Untuk **pertanyaan pertama** Bapak, kenapa tidak menjadi interaktif plotnya disebabkan karena Bapak kurang mendefinisikan suatu label yang berisi method `glue()` kemudian masukan text yang Bapak ingin tampilkan sebagai tooltip. Kemudian setelahnya pada code yang bapak lampirkan, bapak perlu menambahkan parameter `text = label yang telah definisikan sebagai tooltip` di dalam code berikut `geom_col(mapping = aes(fill = freq))`

contoh code:

```
# Pendefinisian label untuk tooltip
freq_egids %>% mutate(label_tooltip = glue(" Endangerment Level : {EGIDS_Label}
     Total Languages: {freq}"

# Membuat visualisasi
lang_stat <- 
ggplot(data = freq_egids, mapping = aes(x = freq, 
                                        y = reorder(EGIDS_Label, freq))) +
  geom_col(aes(fill = freq, text = label_tooltip)) +
  scale_fill_gradient(low = "red", high = "black") +
    labs(title = 'Total of languages in Indonesia based on the endangerment scale' ,
       x = 'Total of languages',
       y = 'Endangerment level') +
  theme_minimal() +
  theme(legend.position = "none")

ggplotly(p = lang_stat, tooltip = "text")
```

Kemudian untuk **pertanyaan kedua**, saran dari saya jangan menggunakan pie chart apabila bapak memiliki kategori nilai yang cukup banyak. Hal ini disebabkan menurut dokumentasi data-to-viz, manusia cenderung lebih sulit melakukan interpretasi sebuah sudut. Sehingga lebih baik dihindari Pak.

Semoga membantu.
Team Algoritma
